pkg/block/namespace_test.go

```diff
@@ -64,35 +70,61 @@ func TestResolveNamespace(t *testing.T) {
 			Name:             "invalid_namespace_wrong_scheme",
 			DefaultNamespace: "memzzzz://foo/",
 			Key:              "bar/baz",
+			Type:             block.IdentifierTypeRelative,
 			ExpectedErr:      block.ErrInvalidNamespace,
 			Expected:         block.QualifiedKey{},
 		},
 		{
 			Name:             "invalid_namespace_invalid_uri",
 			DefaultNamespace: "foo",
 			Key:              "bar/baz",
+			Type:             block.IdentifierTypeRelative,
 			ExpectedErr:      block.ErrInvalidNamespace,
 			Expected:         block.QualifiedKey{},
 		},
 		{
 			Name:             "invalid_key_wrong_scheme",
 			DefaultNamespace: "s3://foo/",
 			Key:              "s4://bar/baz",
+			Type:             block.IdentifierTypeFull,
 			ExpectedErr:      block.ErrInvalidNamespace,
 			Expected:         block.QualifiedKey{},
 		},
+		{
+			Name:             "key_weird_format",
+			DefaultNamespace: "s3://foo/",
+			Key:              "://invalid/baz",
+			Type:             block.IdentifierTypeRelative,
+			Expected: block.QualifiedKey{
+				StorageType:      block.StorageTypeS3,
+				StorageNamespace: "foo",
+				Key:              "://invalid/baz",
+			},
+		},
 	}
 
 	for _, cas := range cases {
-		t.Run(cas.Name, func(t *testing.T) {
-			resolved, err := block.ResolveNamespace(cas.DefaultNamespace, cas.Key)
-			if err != nil && !errors.Is(err, cas.ExpectedErr) {
-				t.Fatalf("got unexpected error :%v - expected %v", err, cas.ExpectedErr)
-			}
-			if cas.ExpectedErr == nil && !reflect.DeepEqual(resolved, cas.Expected) {
-				t.Fatalf("expected %v got %v", cas.Expected, resolved)
+		for _, r := range []block.IdentifierType{cas.Type, block.IdentifierTypeUnknownDeprecated} {
+			relativeName := ""
+			switch r {
+			case block.IdentifierTypeUnknownDeprecated:
+				relativeName = "unknown"
+			case block.IdentifierTypeRelative:
+				relativeName = "relative"
+			case block.IdentifierTypeFull:
+				relativeName = "full"
 			}
-		})
+			t.Run(fmt.Sprintf("%s_%s", cas.Name, relativeName), func(t *testing.T) {
```

Comment:
```suggestion
			t.Run(fmt.Sprintf("%s/%s", cas.Name, relativeName), func(t *testing.T) {
```
will make it more like the regular naming of nested `t.Run`s.  Ideally you would have `t.Run(case.Name)` and nest `t.Run(relativeName)` inside it.  But my 80-column brain can understand why you might wish to avoid that much indentation.